packages/video_player/video_player/test/video_player_test.dart

```diff
@@ -429,7 +431,37 @@ void main() {
         await tester.pumpAndSettle();
 
         expect(controller.value.isPlaying, isFalse);
-        expect(controller.value.position, controller.value.duration);
+        expect(controller.value.position, nonzeroDuration);
+        expect(
+            fakeVideoPlayerPlatform
+                .calls[fakeVideoPlayerPlatform.calls.length - 2],
+            'pause');
+        expect(fakeVideoPlayerPlatform.calls.last, 'seekTo');
```

Comment:
I would prefer to check that the underlying platform actually stopped playing (instead of testing implementation details), but it wasn't clear to me how to do that using the existing test fakes.

If it's worth the effort, I'll dig more into how I might access the underlying platform in unit tests or set up a [golden file](https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package%3Aflutter) test or [device lab](https://github.com/flutter/flutter/wiki/Running-and-writing-tests#running-device-lab-tests-locally) tests for `video_player`. (Those feel rather heavy-handed for this change, but the setup seems like it would be generally useful for `video_player`.)

Reply:
Integration tests in packages/video_player/video_player/example/integration_test/video_player_test.dart looks like it may be a good place to test this. I'll look into it more now.

Reply:
I'll plan to wait until a reviewer has bandwidth to provide guidance.

Here are some things I found in the rabbit hole:
- Running tests in packages/video_player/video_player/example/integration_test/video_player_test.dart with `flutter test` fails (maybe there's a different way to run them?).
- @cyanglaz's TODO to un-skip the test in `example/test_driver/video_player_test.dart` may be resolved (https://github.com/flutter/flutter/issues/43012 was marked fixed in Oct). `flutter drive test_driver/video_player_test.dart` passed once on my machine, but failed with timeouts about 4 times. [The issue](https://github.com/flutter/flutter/issues/43012#issuecomment-548077843) had notes about running these tests on Firebase device lab, so I'm not sure if or where these tests are currently intended to be run.

Reply:
Just a quick note on this question (I haven't looked at the PR in general):

> * Running tests in packages/video_player/video_player/example/integration_test/video_player_test.dart with `flutter test` fails (maybe there's a different way to run them?).

Things in `integration_test` folders are intended to be run via `flutter drive`, using the test driver in `test_driver`; see https://flutter.dev/docs/testing/integration-tests for discussion of the directory structure.

Reply:
Thanks Stuart! I added 2 integration tests. This PR is ready for review.